routing admin/customer basic

phillipet opened the pull request:
Add more  util.getUserType function use for routing admin/customer
Adjust routing customer correctly with new scenario 
Prevent customer use auth-token route to /admin path
Route admin basic ( only handle routing with path = /admin, others admin paths will be implement later )

phillipet committed:
routing admin/customer basic

write isAuth function agaIn return value(false/'a'/'c')
routing  admin/customer basic by auth-token string

phillipet committed:
prevent customer use auth-token route to /admin path

phillipet committed:
keep original isAuth function, add more getUserType function 

 getUserType function  use for Routing

phillipet committed:
remove console log last char auth-token

hemvanh commented on src/router/index.js:
```diff
@@ -22,23 +22,39 @@ export default function({store}) {
+      // user is ADMIN
+    } else if (userType === 'a') {
+      // route to admin home page
+      if (to.path === '/admin') next()
```
if (to.path === '/admin') next()
---> same with:  else next('/admin')
??

hemvanh commented on src/router/index.js:
```diff
@@ -22,23 +22,39 @@ export default function({store}) {
+      // user is ANONYMOUS
+    } else if (userType === false) {
+      // recognize page of user by path to route
+      if (to.path.indexOf('customer') > -1) {
```
what if the url is "/admin/**customers**" ?

phillipet replied:
it go to 404 page http://prntscr.com/l5572p

hemvanh replied:
/admin/customers will be there in the future

hemvanh commented on src/router/index.js:
```diff
@@ -22,23 +22,39 @@ export default function({store}) {
+        } else next('/customer/login')
+      } else {
+        // routing to ADMIN login page
+        if (to.path === '/admin/login') next()
```
if (to.path === '/admin/login') next()
--> same with : **else next('/admin/login')**
??

hemvanh commented on src/util/common.js:
```diff
@@ -63,3 +63,17 @@ export const _procError = err => {
+export function getUserType() {
+  let token = localStorage.getItem('auth-token')
+  let userType = token.substr(token.length - 1)
+  switch (userType) {
```
how about just: return userType  ?

phillipet replied:
ok, I will only return last char, checking char will be handled by router

hemvanh replied:
yea coz userType is null , then !null also = true

hemvanh replied:
so the outside checking we still can keep as the same, no need to change anything

hemvanh reviewed (changes requested):
tune the code abit more

hemvanh reviewed (changes requested):
forgot to update ChangeLogs ?

phillipet committed:
simplitify getUserType, fix some routing 

checking two case '/customer/'  /customer'  prevent route to customer/admin

phillipet committed:
fix checking routing customer no token and admin have token

phillipet committed:
Update change log

phillipet committed:
Update change log, add pull's links of features

phillipet committed:
fix routing algorithm follow test case trello 

#testcase : https://trello.com/c/UdDS06iH/102-routing-admin-check-for-localstorage-auth-token5#comment-5bc07b6a6f3ee22f5d2115e1

phillipet commented:
I done manual testing for new routing algorithm
Test cases at : https://trello.com/c/UdDS06iH/102-routing-admin-check-for-localstorage-auth-token5#comment-5bc07b6a6f3ee22f5d2115e1
